src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -24,96 +37,145 @@
  *
  * See https://github.com/CrossRef/rest-api-doc
  */
-public class CrossRef {
-    private static final Log LOGGER = LogFactory.getLog(CrossRef.class);
+public class CrossRef implements IdParserFetcher<DOI>, EntryBasedParserFetcher, SearchBasedParserFetcher, IdBasedParserFetcher {
```

Comment:
Mhhh....

- IdFetcher: gets an ID for a BibEntry
- EntryBased: gets more information for a BibEntry
- SearchBased: gets BibEntries based on a free-text search query
- IdBased: gets a BibEntry from an identifier

I'm not sure how to simplify this. These interfaces each cover different aspects and not all fetcher have all the capabilities. Do you have any suggestions?